### PR TITLE
Single Node and Multi-Node GPU Docs

### PR DESCRIPTION
**Pull request by vadam5**
Adds documentation for both single and multi-node GPU training in levanter.

### CODE REVIEW

File: docs/Training-On-Your-Data.md [hunk 1]
```diff
@@ -369,9 +369,50 @@ This will spin up a TPU VM instance and install Levanter on it. You can then run
...
+sudo apt-get install cuda-drivers-fabricmanager
+sudo systemctl start nvidia-fabricmanager
+```
+
+If you are using a docker container to train your model, your docker run command should look similar to this
+
+```
+sudo docker run -it --network=host -v ~/src/levanter/cache:/cache -v /home/user/levanter:/levanter --gpus=all --shm-size=16g  ghcr.io/nvidia/jax:levanter
+```
+The main difference between the command here and the one found in the [GPU Docker Development Guide](dev/GPU-Docker-Dev.md) is the `--network=host` argument. This tells the docker container to use the host machine's network instead of the default docker `bridge` network. The `bridge` network is used for communication between docker containers that are on the same host machine, while the `host` network settings is better for container communication across different hosts. Since we are using multiple nodes, we want to use the `host` network setting. Please see the docker's [host](https://docs.docker.com/network/network-tutorial-host/) and [bridge](https://docs.docker.com/network/network-tutorial-standalone/) network documentation for more information.
```
Comment (dlwh) [inline docs/Training-On-Your-Data.md:390]: I don't think this is quite right. host is just the easiest way to do multinode, but there's overlay networks and stuff. I'd just say host networks are the easiest and are probably sufficient for this purpose

File: docs/Training-On-Your-Data.md [hunk 2]
```diff
@@ -369,9 +369,50 @@ This will spin up a TPU VM instance and install Levanter on it. You can then run
...
+- `--trainer.distributed.process_id` - The process ID of the current node. If the node is coordinator for the training run (its IP address was the one specified at `--trainer.distributed.coordinator_address`), its process ID needs to be set to zero. All other nodes in the train run should have a unique integer ID between [1, `num_processes` - 1].
+
+When the above command is run on the coordinator node, it will block until all other processes connect to it. All the other nodes will connect to the coordinator node before they can begin training. All other training run arguments have the same meaning as with single node runs. We recommend thinking about increasing your `--trainer.train_batch_size` value when you scale from single node to multi-node training, as this is the global batch size for your training job and you've now increased your compute capacity.
+
+
+### Switching Between GPU and TPU
+In levanter, you can switch between using TPUs and GPUs in the middle of a training run. See our tutorial on [Switching Hardware Mid-Training Run](Hardware-Agnostic-Training.md) to learn more.
```
Comment (dlwh) [inline docs/Training-On-Your-Data.md:415]:
```suggestion
In Levanter, you can switch between using TPUs and GPUs in the middle of a training run. See our tutorial on [Switching Hardware Mid-Training Run](Hardware-Agnostic-Training.md) to learn more.
```

File: docs/Training-On-Your-Data.md [hunk 3]
```diff
@@ -369,9 +369,50 @@ This will spin up a TPU VM instance and install Levanter on it. You can then run
 gcloud compute tpus tpu-vm ssh my-tpu   --zone us-east1-d --worker=all --command="WANDB_API_KEY=... levanter/infra/launch.sh python levanter/src/levanter/main/train_lm.py --config_path gs://path/to/config.yaml"
 ```
 
-### GPU
 
-TODO, but you can mostly follow the guide for [TPU](#tpu) above.
+### GPU Training
```
Comment (dlwh) [inline docs/Training-On-Your-Data.md:373]: I think this should go in the GPU guide actually. We should also say that if you're SLURM (and using Pyxis), you won't need to do provide the distributed arguments. JAX/Levanter will infer them for you.

**Review by dlwh — CHANGES_REQUESTED** [review]
Couple of thoughts/corrections. Thanks!

File: docs/Getting-Started-GPU.md [hunk 4]
```diff
@@ -265,16 +303,15 @@ Something is wrong and this doesn't work right now on the NLP cluster. In theory
...
-
-srun --nodes=2 python -m levanter.main.train_lm --config config/gpt2_small.yaml --trainer.per_device_parallelism -1
+CONTAINER_PATH="ghcr.io/nvidia/jax:levanter"
+TRAINING_COMMAND="python -m levanter.main.train_lm --config_path config/gpt2_7b.yaml --trainer.ray.auto_start_cluster false --trainer.per_device_parallelism -1"
 
-Submit the job with sbatch:
-
-```bash
-sbatch my-job.sh
+srun docker run --gpus=all --shm-size=16g --rm $CONTAINER_PATH $TRAINING_COMMAND
```
Comment (vadam5) [inline docs/Getting-Started-GPU.md:309]: Pretty sure srun + docker run is wrong. Need to test on the NLP cluster.

Comment (dlwh) [reply docs/Getting-Started-GPU.md:309]: huh i thought i had tested that, but i dunno

**Review by dlwh — APPROVED** [review]
lgtm thanks!

**Pull request merged**